common4j/src/main/com/microsoft/identity/common/java/jwt/JwtUtils.java

```diff
@@ -59,5 +63,38 @@ public static String generateJWT(@NonNull final JwtRequestHeader header,
                 Base64Util.encodeUrlSafeString(headerJson.getBytes(ENCODING_UTF8)) + "." + Base64Util.encodeUrlSafeString(bodyJson.getBytes(ENCODING_UTF8));
         return encodedJwt;
     }
+
+    /**
+     * Generate a JWT request header to be used in MSA DR flows.
+     * @param keyContext key context to include in header
+     * @return a request headers object
+     */
+    @NonNull
+    public static JwtRequestHeader generateJwtRequestHeaderForMsaDR(final byte[] keyContext) {
+        final JwtRequestHeader jwtRequestHeader = new JwtRequestHeader();
+        jwtRequestHeader.setType();
+        jwtRequestHeader.setAlg(JwtRequestHeader.ALG_VALUE_HS256);
+        jwtRequestHeader.setKId(JwtRequestHeader.KID_VALUE_ECDH);
+        jwtRequestHeader.setCtx(Arrays.toString(keyContext));
+
+        return jwtRequestHeader;
+    }
+
+    /**
+     * Generate a JWT request body to be used in MSA DR flows.
+     * @param deviceToken device token to include in body
+     * @return a request body object
+     */
+    @NonNull
+    public static JwtRequestBody generateJwtRequestBodyForMsaDR(final String audienceUrl, final String nonce, final String deviceToken) {
+        final JwtRequestBody jwtRequestBody = new JwtRequestBody();
+        jwtRequestBody.setAudience(audienceUrl); //TODO: Can i hard code this to login.microsoftonline.com instead of passing parameter?
```

Comment:
> //TODO: Can i hard code this to login.microsoftonline.com instead of passing parameter?

What will be the benefit?

Reply:
I don't remember to be honest, this was like a month ago. probably no longer relevant